main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"code-review-bot/internal"
+	"code-review-bot/internal/web"
+	"code-review-bot/internal/web/routes"
+	"context"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v68/github"
+	"github.com/gregjones/httpcache"
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/openai/openai-go"
+	"github.com/openai/openai-go/option"
+	"go.uber.org/zap"
+)
+
+func initializeGithubClient(ctx context.Context, githubToken string, logger *zap.Logger) *github.Client {
+	client := github.NewClient(httpcache.NewMemoryCacheTransport().Client()).WithAuthToken(githubToken)
+
+	user, _, err := client.Users.Get(ctx, "")
+	if err != nil {
+		logger.Fatal("Failed to authenticate with GitHub API", zap.Error(err))
+	}
```

Comment:
While the context is created in `main` and passed along, it would be ideal to ensure that the context has a deadline or cancellation mechanism tied to the lifecycle of the request. This prevents resource leaks in case of long-running operations or client disconnections.